### PR TITLE
ZEPPELIN-1613. PerUser scoped mode doesn't work

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?
For perUser scoped mode, difference users still use the same interpreter instance which is incorrect. This is due to they are using the same interpreter instance key.  This PR will fix this by using different interpreter instance key for different users. Besides, I also did some refactoring and add more log.

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-1613

### How should this be tested?
Tested manually. 

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
\cc @jongyoul @Leemoonsoo  Please help review.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java [hunk 1]
```diff
@@ -1172,28 +1172,28 @@ private String getInterpreterInstanceKey(String user, String noteId, Interpreter
     String key;
     if (option.isExistingProcess()) {
       key = Constants.EXISTING_PROCESS;
-    } else if (!option.perNoteShared()) {
-      key = noteId;
-      if (shiroEnabled && !option.perUserShared()) {
-        key = user + ":" + key;
-      }
+    } else if (option.perNoteScoped()) {
```
Comment (zjffdu) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java]: hmm, right, I think I miss that combination, will update the PR soon.

**Comment by prabhjyotsingh** [comment]
Thanks for creating this issue, this wasn't working for me either.

**Comment by zjffdu** [comment]
@jongyoul I fix the combination issue and also did some refactoring. 
* rename interpreterInstanceKey to sessionKey

**Comment by zjffdu** [comment]
@jongyoul @Leemoonsoo Could you help review ?

**Comment by Leemoonsoo** [comment]
@zjffdu Looks great to me!
Merge to master if there're no further comments.

**Pull request closed without merging**